src/split_reduce.cpp

```diff
@@ -162,19 +180,20 @@ void split_reduce::apply(module_pass_manager& mpm) const
         auto* rm = ins->module_inputs().front();
         if(get_reduce_size(rm) < split_size)
             continue;
-        auto splits = find_split(rm);
+        splitter s{rm};
+        auto splits = s.find_splits();
         if(splits.empty())
             continue;
         // Only use split reduce with float for now
         // TODO: Support half and other data types
         if(not std::all_of(splits.begin(), splits.end(), [](instruction_ref split) {
-               return split->get_shape().type() == shape::float_type;
+               return contains({shape::float_type, shape::half_type}, split->get_shape().type());
```

Comment:
update TODO above to remove "half" type.

Reply:
can you remove TODO for the Half ?

Reply:
@pfultz2

Reply:
Updated.